qiskit_nature/circuit/library/initial_states/hartree_fock.py

```diff
@@ -12,63 +12,79 @@
+
 
 class HartreeFock(QuantumCircuit):
     """A Hartree-Fock initial state."""
 
     def __init__(self,
-                 num_orbitals: int,
-                 num_particles: Union[Tuple[int, int], int],
-                 qubit_mapping: str = 'parity',
-                 two_qubit_reduction: bool = True,
-                 sq_list: Optional[List[int]] = None) -> None:
+                 num_spin_orbitals: int,
+                 num_particles: Tuple[int, int],
+                 qubit_converter: QubitConverter) -> None:
         """
         Args:
-            num_orbitals: The number of spin orbitals, has a min. value of 1.
-            num_particles: The number of particles. If this is an integer, it is the total (even)
-                number of particles. If a tuple, the first number is alpha and the second number is
-                beta.
-            qubit_mapping: The mapping type for qubit operator.
-            two_qubit_reduction: A flag indicating whether or not two qubit is reduced.
-            sq_list: The position of the single-qubit operators that anticommute with the cliffords.
+            num_spin_orbitals: The number of spin orbitals, has a min. value of 1.
+            num_particles: The number of particles as a tuple storing the number of alpha- and
+                           beta-spin electrons in the first and second number, respectively.
+            qubit_converter: a QubitConverter instance.
         """
+
         # get the bitstring encoding the Hartree Fock state
-        bitstr = hartree_fock_bitstring(num_orbitals, num_particles, qubit_mapping,
-                                        two_qubit_reduction, sq_list)
+        bitstr = hartree_fock_bitstring(num_spin_orbitals, num_particles)
+
+        # encode the bitstring as a `FermionicOp`
+        label = ['+' if bit else 'I' for bit in bitstr]
+        bitstr_op = FermionicOp(''.join(label))
+
+        # map the `FermionicOp` to a qubit operator
+        qubit_op = qubit_converter.to_qubit_ops([bitstr_op])[0]
 
         # construct the circuit
-        qr = QuantumRegister(len(bitstr), 'q')
+        qr = QuantumRegister(qubit_op.num_qubits, 'q')
         super().__init__(qr, name='HF')
 
-        # add gates in the right positions
-        for i, bit in enumerate(reversed(bitstr)):
+        # Add gates in the right positions: we are only interested in the `X` gates because we want
+        # to create particles (0 -> 1) where the initial state introduced a creation (`+`) operator.
+        for i, bit in enumerate(qubit_op.primitive.table.X[0]):
             if bit:
                 self.x(i)
 
 
-def hartree_fock_bitstring(num_orbitals: int,
-                           num_particles: Union[Tuple[int, int], int],
-                           qubit_mapping: str = 'parity',
-                           two_qubit_reduction: bool = True,
-                           sq_list: Optional[List[int]] = None) -> List[bool]:
+def hartree_fock_bitstring(num_spin_orbitals: int,
```

Comment:
Maybe we should think about the name here too - I think of a bitstring as a string of 0's and 1's  '010101111' which perhaps this was at some point. Anyway when we figure the QubitCOnvertor TODO maybe we can consider a better name.

Reply:
Right, I had kind of ignored that the name no longer fits so nicely.. But I agree, let's do that when we tackle the other TODO :+1: